common/src/api/external/mod.rs

```diff
@@ -1269,6 +1278,15 @@ impl JsonSchema for IpNet {
     }
 }
 
+/// A `VpcAddress` represents either an explicit IP network (v4 or v6)
+/// or an entire VPC (identified by its VNI).
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, JsonSchema)]
+#[serde(tag = "type", content = "value", rename_all = "snake_case")]
+pub enum VpcAddress {
```

Comment:
I think I understand why this enum exists, to cover the ways we ascribe a firewall rule to a set of hosts. Would a name more like `HostIdentifier` be appropriate?

I also wonder whether this needs to live in the external API. There are no external calls that should see this, AFAICT, so `common/src/api/internal/mod.rs` might be more appropriate. There may be some Rust visibility or Cargo dependency that makes that less attractive, but it might be nice to investigate.

Reply:
`HostIdentifier` is a great suggestion, thanks! And good call on the internal API; I always forget that we have that. Renamed and moved in 47361a5a.

(Minor follow-up question: after much playing around with building, rebuilding, and running the various `_openapi` tests with `EXPECTORATE=overwrite`, I still can't figure out why `HostIdentifier` shows up in `sled-agent.json` but not `nexus-internal.json`. The build seems happy, so I'm not going to sweat it, but I'm still a bit confused.)

Reply:
That might be a question for @ahl! It's possible that types only show up in the OpenAPI output when they are actually needed to make requests to a given server. Since this type is technically only used by Nexus making calls _to_ the sled agent, only including needed names could explain it. But that's a WAG :)